tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -418,8 +418,20 @@ def test_pantsd_multiple_parallel_runs(self):
       self.assert_success(creator_handle.join())
       self.assert_success(waiter_handle.join())
 
-  def _assert_pantsd_keyboardinterrupt_signal(self, signum, messages, regexps=[],
+  def _assert_pantsd_keyboardinterrupt_signal(self, signum,
+                                              single_msg=None, messages=[], regexps=[],
```

Comment:
`single_msg` doesn't seem to pull its weight vs just having the caller wrap a string in a list... Most of the callers still use a single-element `messages`, but having both here opens up:
1. Questions of whether `single_msg` and `messages` are mutually exclusive options
2. Makes `messages` have a default mutable value, which can lead to errors

I'd probably ditch `single_msg` :)

Reply:
Agreed!

Reply:
Well, `single_msg` is intended to make it easier to check whether a message matches the output *exactly*. But this can be done with regexps anyway (actually, the whole thing can be done with regexps anyway, if we force the use of `re.escape()`). I will try narrowing this to just allow `regexps` and see if that makes things confusing (since this method is only being called in a few places in this file).